xmodule/modulestore/mixed.py

```diff
@@ -797,6 +799,13 @@ def delete_item(self, location, user_id, **kwargs):  # lint-amnesty, pylint: dis
         Delete the given item from persistence. kwargs allow modulestore specific parameters.
         """
         store = self._verify_modulestore_support(location.course_key, 'delete_item')
+        # .. event_implemented_name: XBLOCK_DELETED
+        XBLOCK_DELETED.send_event(
+            xblock_info=XBlockData(
+                usage_key=location,
+                block_type=location.block_type,
+            )
+        )
         return store.delete_item(location, user_id=user_id, **kwargs)
```

Comment:
Should we do something like this to be precise about the event.

```
item = store.delete_item(location, user_id=user_id, **kwargs)
# .. event implemented_name: XBLOCK_DELETED
<event here>
return item
```

Reply:
@tecoholic Makes sense.